src/credentials/vue/authForms/manageBuilderId.vue

```diff
@@ -0,0 +1,164 @@
+<template>
+    <div class="auth-form container-background border-common" id="builder-id-form">
+        <div v-show="canShowAll">
+            <FormTitle :isConnected="isConnected">AWS Builder ID</FormTitle>
+
+            <div v-if="stage === 'START'">
+                <div class="form-section">
+                    <div>
+                        With AWS Builder ID, sign in for free without an AWS account.
+                        <a href="https://docs.aws.amazon.com/signin/latest/userguide/sign-in-aws_builder_id.html"
+                            >Read more.</a
+                        >
+                    </div>
+                </div>
+
+                <div class="form-section">
+                    <button v-on:click="startSignIn()">Sign up or Sign in</button>
+                </div>
+            </div>
+
+            <div v-if="stage === 'WAITING_ON_USER'">
+                <div class="form-section">
+                    <div>Follow instructions...</div>
+                </div>
+            </div>
+
+            <div v-if="stage === 'CONNECTED'">
+                <div class="form-section">
+                    <div v-on:click="signout()" style="cursor: pointer; color: #75beff">Sign out</div>
+                </div>
+            </div>
+        </div>
+    </div>
+</template>
+<script lang="ts">
+import { PropType, defineComponent } from 'vue'
+import BaseAuthForm from './baseAuth.vue'
+import FormTitle from './formTitle.vue'
+import { AuthFormId, AuthStatus } from './shared.vue'
+import { WebviewClientFactory } from '../../../webviews/client'
+import { AuthWebview } from '../show'
+
+const client = WebviewClientFactory.create<AuthWebview>()
+
+export default defineComponent({
+    name: 'CredentialsForm',
+    extends: BaseAuthForm,
+    components: { FormTitle },
+    props: {
+        state: {
+            type: Object as PropType<BaseBuilderIdState>,
+            required: true,
+        },
+    },
+    data() {
+        return {
+            stage: stages.START as BuilderIdStage,
+            isConnected: false,
+            builderIdCode: '',
+            canShowAll: false,
+        }
+    },
+    async created() {
+        await this.update()
+        this.canShowAll = true
+    },
+    methods: {
+        async startSignIn() {
+            this.stage = stages.WAITING_ON_USER
+            await this.state.startBuilderIdSetup()
+            await this.update()
+        },
+        async update() {
+            this.stage = await this.state.stage()
+            this.isConnected = await this.state.isAuthConnected()
+            this.emitAuthConnectionUpdated(this.state.id)
+        },
+        async signout() {
+            await this.state.signout()
+
+            this.update()
+        },
+    },
+})
+
+export const stages = {
+    START: 'START',
+    WAITING_ON_USER: 'WAITING_ON_USER',
+    CONNECTED: 'CONNECTED',
+} as const
+
+type BuilderIdStage = (typeof stages)[keyof typeof stages]
```

Comment:
There is little reason to create these pseudo-enums:
```suggestion
type BuilderIdStage = 'START' | 'WAITING_ON_USER' | 'CONNECTED'
```

Reply:
The reason behind this was to use `stages.START` instead of `'START'` so if I need to refactor or just finding all uses of `START`. Though I wasn't always following this rule when I should have, but I've updated to use these properties instead of strings.

Reply:
Update: Fixed this in a future PR that includes other refactors